async/src/main/java/org/odk/collect/async/network/ConnectivityProvider.kt

```diff
@@ -0,0 +1,24 @@
+package org.odk.collect.async.network
+
+import android.content.Context
+import android.net.ConnectivityManager
+import org.odk.collect.async.Scheduler
+
+class ConnectivityProvider(private val context: Context) : NetworkStateProvider {
```

Comment:
We'll need to replace this implementation at some point, as it's pretty much all deprecated.